torch/utils/backend_registration.py

```diff
@@ -82,20 +82,140 @@ def _check_register_once(module, attr):
         raise RuntimeError(f"The custom device module of {module} has already been registered with {attr}")
 
 
-def generate_methods_for_privateuse1_backend() -> None:
+def _normalization_device(custom_backend_name: str, device: Optional[Union[int, torch.device]] = None) -> int:
+    get_device_idx = "get_current_device_index"
+    if device is None:
+        if hasattr(torch, custom_backend_name) and \
+                hasattr(f'torch.{custom_backend_name}', get_device_idx):
+            device_idx = getattr(f'torch.{custom_backend_name}',
+                                 get_device_idx)()
+        else:
+            # The default scene device index is 0.
+            device_idx = 0
+    elif isinstance(device, torch.device):
+        if device.type != custom_backend_name:
+            raise RuntimeError(f"Invalid device, must be {custom_backend_name} device")
+        else:
+            device_idx = device.index
+    return device_idx
+
+
+def _generate_tensor_methods_for_privateuse1_backend(custom_backend_name: str) -> None:
+    @property  # type: ignore[misc]
+    def wrap_tensor_backend(self: torch.Tensor) -> bool:
+        return self.device.type == custom_backend_name
+
+    _check_register_once(torch.Tensor, f'is_{custom_backend_name}')
+    setattr(torch.Tensor, f'is_{custom_backend_name}', wrap_tensor_backend)
+
+    def wrap_tensor_to(self: torch.Tensor, device: Optional[Union[int, torch.device]] = None, non_blocking=False,
```

Comment:
Can we add docstrings to all of these functions we're adding so that at least `?torch.Tensor.foo` will work (even though it is not on the website doc).

Reply:
I have referenced the existing comments to supplement the corresponding doc for the generated method.